### PR TITLE
Blog app - views

### PR DESCRIPTION
**Pull request by devendra-alt**
## [Blog app - views](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md)

- Added view for `/`
- Add users list for `/users`
- Add single user view on `/users/10`
- Add post list for single users at `/users/10/posts`
- Add single post display on `/users/10/posts/20`

### please use the following `id` and `URL` to get the expected results
for users => `/users`
for specific user => `/users/10`
for user post lists => `/users/10/posts`
for specific post page => `/users/10/posts/20`

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1 +1,14 @@
...
+<div id="user-profile">
+  <%= render partial:'user', locals:{user: @user} %>
+  <div class="user-bio">
+    <h2>Bio</h2>
+    <%= @user.bio %>
+  </div>
+  <div class="user-profile">
+    <% if @user.posts_count.positive?%>
+      <% @posts.each_with_index do |post,index| %>
+        <%= render partial: 'posts/post', locals:{post:post,index:index} %>
```
Comment (rloterh) [inline app/views/users/show.html.erb]: - [x] For this activity we want you to use the `recent_post` method created in previous lesson to render the user's 3 most recent post. It should like something like below:
```ruby
# app/views/users/show.html.erb

[...]

<% @user.recent_post.each do |post| %>

  <h4><%= link_to post.title, user_post_path(post) %></h4>
  <p><%= post.text %></p>
  <p>
    <small>Comments: <%= post.comments_counter || 0 %>,</small> 
    <small>Likes: <%= post.likes_counter || 0 %></small>
  </p>

<% end %>

<button>
  <%= link_to 'See all Post', user_posts_path(@user) %>
</button>
```

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -1 +1,9 @@
-<h1>Posts<h1>
\ No newline at end of file
+<div id="user-posts">
+  <%= render partial:'users/user',locals:{user:@user} %>
+  <% @posts.each_with_index do |post,index| %>
+    <%= render partial: 'posts/post', locals:{post:post,index:index} %>
+    <%= render partial: 'comments/comment', locals:{comments:@comments,users:@users}%>
+  <%end%>
+  <div class="pagination"><%= will_paginate @posts %></div>
+</div>
```
Comment (rloterh) [inline app/views/posts/index.html.erb]: - [x] Similarly, let's use the`post.recent_comments` created in previous lesson to render 5 most recent comments in the `app/views/posts/index.html.erb`, which should look like below:

```ruby
# app/views/posts/index.html.erb

<% post.recent_comments.each do |comment| %>
  <p><%= comment.author.name %>: <%= comment.text %></p> 
 <% end %>
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Devendra, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🌴 🚦 🚗 🏫 🏡 _🏃

## Highlights 👍  
- Good job on view creation so far!
- ReadMe looks great!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Danieldotcomcoder — APPROVED** [review]
Hello @devendra-alt , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 👍 👍 😄 👏 

Congratulations! 🎉 🎉 🎉 

## Highlights
- Professional Readme File ✔✔
- Clear and nice code ✔✔

comp-2
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**